tests/runtests.pl

```diff
@@ -1070,6 +1073,8 @@ sub getrunnerlogdir {
 sub singletest_shouldrun {
     my $testnum = $_[0];
     my $why;   # why the test won't be run
+    my $why_keywords; # why the test won't be run due to keywords
```

Comment:
I'm not sure it's worthwhile keeping track of all the ways a test won't be run. It's fine to me to check all the criteria in order and and just mention the first one discovered to cause it to be skipped. This will simplify the code.

Reply:
That makes sense. I initially implemented it this way to ensure all skip reasons were properly reported, but if you think it’s not necessary, I’m happy to simplify the code.